pkg/ipam/crd.go

```diff
@@ -549,14 +549,38 @@ func (n *nodeStore) isIPInReleaseHandshake(ip string) bool {
 }
 
 // allocateNext allocates the next available IP or returns an error
-func (n *nodeStore) allocateNext(allocated ipamTypes.AllocationMap, family Family) (net.IP, *ipamTypes.AllocationIP, error) {
+func (n *nodeStore) allocateNext(allocated ipamTypes.AllocationMap, family Family, owner string) (net.IP, *ipamTypes.AllocationIP, error) {
 	n.mutex.RLock()
 	defer n.mutex.RUnlock()
 
 	if n.ownNode == nil {
 		return nil, nil, fmt.Errorf("CiliumNode for own node is not available")
 	}
 
+	// Check if IP has a custom owner (only supported in manual CRD mode)
+	if n.conf.IPAMMode() == ipamOption.IPAMCRD && len(owner) != 0 {
+		for ip, ipInfo := range n.ownNode.Spec.IPAM.Pool {
+			if ipInfo.Owner == owner {
+				parsedIP := net.ParseIP(ip)
+				if parsedIP == nil {
+					log.WithFields(logrus.Fields{
+						fieldName: n.ownNode.Name,
+						"ip":      ip,
+					}).Warning("Unable to parse IP in CiliumNode custom resource")
+					return nil, nil, fmt.Errorf("invalid custom ip %s for %s. ", ip, owner)
+				}
+				if DeriveFamily(parsedIP) != family {
+					log.WithFields(logrus.Fields{
+						fieldName: n.ownNode.Name,
+						"ip":      ip,
+					}).Warning("Custom ip family do not match, skip. ")
```

Comment:
I don't think you want to warn in this scenario. Consider dual-stack setups, where one might have a pool definition like this: 

```yaml
    ipam:
      pool:
        10.244.0.1:
          owner: default/xwing
        10.244.0.2: {}
        10.244.0.3: {}
        10.244.0.4: {}
        'fd02:1::1':
          owner: default/xwing
        'fd02:1::2': {}
        'fd02:1::3': {}
        'fd02:1::4': {}
```

When requesting the IPv6 address, it would first observe the IPv6 address, warn that the IP family does not match, but then assign the IPv4 regardless. That means that the warning is pointless.

Therefore I don't think you want to warn here.

Reply:
I don't understand enough, thank you